src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6355,6 +6408,416 @@ register_handler(int sig, struct sigaction *old_handler)
+
+	if (next_io_setup == NULL) {
+		next_io_setup = dlsym(RTLD_NEXT, "io_setup");
+		D_ASSERT(next_io_setup != NULL);
+	}
+
+	rc = next_io_setup(maxevents, ctxp);
+	if (rc < 0)
+		return rc;
+
+	D_ALLOC(aio_ctx_obj, sizeof(aio_ctx_t));
+	if (aio_ctx_obj == NULL) {
+		io_destroy(*ctxp);
+		return -ENOMEM;
+	}
+	aio_ctx_obj->ctx   = *ctxp;
+	aio_ctx_obj->depth = maxevents;
+	atomic_store_relaxed(&aio_ctx_obj->num_op_submitted, 0);
+	atomic_store_relaxed(&aio_ctx_obj->num_op_done, 0);
+	aio_ctx_obj->inited       = false;
+	aio_ctx_obj->on_dfs       = false;
+	aio_ctx_obj->ev_done_list = NULL;
+	aio_ctx_obj->ev_done_h    = 0;
+	aio_ctx_obj->ev_done_t    = 0;
+	aio_ctx_obj->ev_queue_len = 0;
+	*ctxp                     = (io_context_t)aio_ctx_obj;
+
+	rc = D_MUTEX_INIT(&aio_ctx_obj->lock, NULL);
+	if (rc) {
+		io_destroy(*ctxp);
+		D_FREE(aio_ctx_obj);
+		return -daos_der2errno(rc);
+	}
+
+	/* assume all io requests are over DFS for now. */
+	rc = create_ev_eq_for_aio(aio_ctx_obj);
+	if (rc) {
+		return (-rc);
+	}
+
+	return 0;
+}
+
+int
+io_destroy(io_context_t ctx)
+{
+	aio_ctx_t   *aio_ctx_obj = (aio_ctx_t *)ctx;
+	io_context_t ctx_save    = aio_ctx_obj->ctx;
+
+	if (next_io_destroy == NULL) {
+		next_io_destroy = dlsym(RTLD_NEXT, "io_destroy");
+		D_ASSERT(next_io_destroy != NULL);
+	}
+	D_MUTEX_DESTROY(&aio_ctx_obj->lock);
+	D_FREE(aio_ctx_obj->ev_done_list);
+	D_FREE(aio_ctx_obj);
+	return next_io_destroy(ctx_save);
+}
+
+static int
+create_ev_eq_for_aio(aio_ctx_t *aio_ctx)
+{
+	int rc, i, j, idx;
+	int num_aio_eq_free, num_aio_eq_create;
+
+	D_MUTEX_LOCK(&lock_aio);
+	num_aio_eq_free   = (int)eq_count_max - (int)eq_count - (int)aio_eq_count;
+	num_aio_eq_create = min(aio_ctx->depth, num_aio_eq_free);
+	if (num_aio_eq_create > 0) {
+		/* allocate EQs for aio context*/
+		for (i = 0; i < num_aio_eq_create; i++) {
+			idx = aio_eq_count;
+			rc  = daos_eq_create(&aio_eq_list[idx + i].eq);
+			if (rc)
+				goto free_eq;
+			D_MUTEX_INIT(&aio_eq_list[idx + i].lock, NULL);
+			atomic_store_relaxed(&aio_eq_list[idx + i].n_op_queued, 0);
+			atomic_store_relaxed(&aio_eq_list[idx + i].n_op_done, 0);
+		}
+		aio_eq_count += num_aio_eq_create;
+	}
+	aio_ctx->first_eq = depth_sum_accu % aio_eq_count;
+	depth_sum_accu += aio_ctx->depth;
+	D_MUTEX_UNLOCK(&lock_aio);
+
+	if (aio_eq_count == 0) {
+		DS_ERROR(EBUSY, "no EQs created for AIO contexts");
+		return EBUSY;
+	}
+
+	D_MUTEX_LOCK(&aio_ctx->lock);
+	if (aio_ctx->inited == false) {
+		D_ALLOC_ARRAY(aio_ctx->ev_done_list, aio_ctx->depth);
+		if (aio_ctx->ev_done_list == NULL) {
+			D_MUTEX_UNLOCK(&aio_ctx->lock);
+			return ENOMEM;
+		}
+		/* empty queue */
+		aio_ctx->ev_done_h       = 0;
+		aio_ctx->ev_done_t       = 0;
+		aio_ctx->ev_done_list[0] = NULL;
+		aio_ctx->ev_queue_len    = 0;
+		aio_ctx->inited          = true;
+	}
+	aio_ctx->on_dfs = true;
+	D_MUTEX_UNLOCK(&aio_ctx->lock);
+
+	return 0;
+
+free_eq:
+	for (j = 0; j < i; j++) {
+		daos_eq_destroy(aio_eq_list[idx + j].eq, 0);
```

Comment:
check error condition for coverity

Reply:
Added the check of returned value for daos_eq_destroy(). Thank you!